src/storagemon/storagemon.cpp

```diff
@@ -0,0 +1,129 @@
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA
+ *
+ */
+
+#include <QCoreApplication>
+#include <QDebug>
+#include <QTimer>
+#include <QTranslator>
+
+#include "creporterapplicationsettings.h"
+#include "creportercoreregistry.h"
+#include "creporternamespace.h"
+#include "creporternotification.h"
+#include "creportersavedstate.h"
+#include "creporterutils.h"
+
+#ifndef QT_NO_DEBUG_OUTPUT
+#include "creporterlogger.h"
+#endif
+
+constexpr int TOO_MUCH_BYTES = 100 << 20;
+
+using CReporter::LoggingCategory::cr;
+
+/*!
+  * @brief Crash Reporter storage usage monitor application.
+  */
+class Application : public QCoreApplication
+{
+    Q_OBJECT
+
+public:
+    /*!
+      * @brief Constructor
+      *
+      * @param argc Argument count.
+      * @param argv Arguments.
+      */
+    Application(int argc, char **argv)
+        : QCoreApplication(argc, argv)
+    {
+        QTranslator *translator = new QTranslator(this);
+        translator->load("crash-reporter_eng_en", "/usr/share/translations");
+        installTranslator(translator);
+
+        QTimer::singleShot(0, this, &Application::run);
+    }
+
+    ~Application()
+    {
+        CReporterApplicationSettings::freeSingleton();
+    }
+
+private:
+    void run()
+    {
+        checkStorageUsage();
+        quit();
+    }
+
+    void checkStorageUsage()
+    {
+
+        qint64 totalSize = 0;
+        CReporterCoreRegistry *registry = CReporterCoreRegistry::instance();
+        foreach (const QString &coreFile, registry->collectAllCoreFiles()) {
+            totalSize += QFileInfo(coreFile).size();
+        }
+
+        if (totalSize < TOO_MUCH_BYTES) {
+            qCDebug(cr) << "Low storage usage - will not be reported";
+            return;
+        }
+
+        qCDebug(cr) << "High storage usage - will be reported";
+
+        CReporterSavedState *state = CReporterSavedState::instance();
+
+        CReporterNotification notification(CReporter::StorageUsageNotificationEventType,
+                state->storageUsageNotificationId(), this);
+
+        //% "Many crash reports available"
+        notification.update(
+                qtTrId("crash_reporter-notify-storage_usage"),
+                //% "You should upload or delete some of them. You have %1 MiB of crash data."
+                qtTrId("crash_reporter-notify-storage_usage_body").arg(totalSize >> 20));
+
+        state->setStorageUsageNotificationId(notification.id());
+
+        CReporterSavedState::freeSingleton();
+    }
+
+};
+
+/*!
+  * @brief Crash Reporter storage usage monitor main function.
+  *
+  * @param argc Argument count.
+  * @param argv Arguments.
```

Comment:
C++ programmer should really know argc and argv :) Obvious comments are more noise than signal and don't think we need to generate documentation for something that isn't a library anyway.

Reply:
Just following the conventions used elsewhere

Reply:
Let's not keep repeating conventions that don't make sense :) I'm all for removing similar lines elsewhere.

Reply:
Dropped